monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.ts.micrometer.prometheus;
+
+import static io.restassured.RestAssured.given;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+public class HttpPathMetricsPatternIT {
+
+    private static final int ASSERT_METRICS_TIMEOUT_MINUTES = 1;
+    private static final List<String> HTTP_SERVER_REQUESTS_METRICS_SUFFIX = Arrays.asList("count", "sum", "max");
+    private static final String HTTP_SERVER_REQUESTS_METRICS_FORMAT = "http_server_requests_seconds_%s{method=\"GET\",outcome=\"%s\",status=\"%d\",uri=\"%s\"}";
+    private static final String REDIRECT_ENDPOINT = "/test/redirect";
+    private static final String NOT_FOUND_ENDPOINT = "/test/not-found";
+    private static final String MOVED_ENDPOINT = "/test/moved/{id}";
+    private static final String EMPTY_ENDPOINT = "/test/";
+
+    @QuarkusApplication
+    static RestService app = new RestService()
+            .withProperty("quarkus.management.enabled", "true");
+
+    @Test
+    public void testNotFoundEndpoint() {
+        whenCallNotFoundEndpoint();
+        thenMetricIsExposedInServiceEndpoint(404, "CLIENT_ERROR", NOT_FOUND_ENDPOINT);
```

Comment:
```suggestion
        thenMetricIsExposedInServiceEndpoint(404, "CLIENT_ERROR", "NOT_FOUND");
```
because actual response is `http_server_requests_seconds_count{method="GET",outcome="CLIENT_ERROR",status="404",uri="NOT_FOUND"} 1.0` and `/test/not-found` has no path param. I suggest to complement this test with `testNotFoundEndopintDynamic` (using your terminology, not sure it's alright TBH!) so that you can test `NOT_FOUND_ENDPOINT` in URI.

Reply:
Currently with the Quarkus code, for 404, as you mentioned the response is `http_server_requests_seconds_count{method="GET",outcome="CLIENT_ERROR",status="404",uri="NOT_FOUND"} 1.0 `  and I've tried diferent ways but not sure if it's possible to do coincide them:  `pathInfo` with `pathInfo` for returning the uri path , according source code:
` } else if (code == 404) {
                if (isTemplatedPath(pathInfo, initialPath)) {
                    return Tag.of("uri", pathInfo);
                } else {
                    return URI_NOT_FOUND;
                }
            }.`

Reply:
I don't know as well. but if you are right, this is the dead code which is not correct. I suppose it is possible, but better check with the developer.

Reply:
ok I've just discovered how it could work, I've pushed my commit right now.